src/talk/renderer/UserStatus/components/UserStatusForm.vue

```diff
@@ -0,0 +1,187 @@
+import { translate as t } from '@nextcloud/l10n'
+import { useUserStatusStore } from '../userStatus.store.js'
+import UserStatusFormClearAt from './UserStatusFormClearAt.vue'
+import UserStatusFormCustomMessage from './UserStatusFormCustomMessage.vue'
+import UserStatusFormStatusType from './UserStatusFormStatusType.vue'
+import { convertPredefinedStatusToUserStatus } from '../userStatus.utils.js'
+import UserStatusFormPredefinedOption from './UserStatusFormPredefinedOption.vue'
+
+const emit = defineEmits(['submit'])
+
+const userStatusStore = useUserStatusStore()
+
+/** @type {import('vue').Ref<import('./userStatus.types.ts').UserStatus>} */
+const userStatus = ref({ ...userStatusStore.userStatus })
+
+const backupStatus = ref(userStatusStore.backupStatus ? { ...userStatusStore.backupStatus } : null)
+
+const isDirty = ref(false)
+
+const { predefinedStatuses } = storeToRefs(userStatusStore)
+
+const statusIsUserDefined = computed(() => userStatus.value.icon || userStatus.value.message)
+
+const patchStatus = (newUserStatus) => {
+	isDirty.value = true
+	backupStatus.value = null
+	Object.assign(userStatus.value, newUserStatus)
+}
+
+const selectPredefinedStatus = (status) => {
+	patchStatus(convertPredefinedStatusToUserStatus(status))
+}
+
+const clearUserStatusCustomMessage = () => {
+	patchStatus({
+		messageId: null,
+		messageIsPredefined: false,
+		message: null,
+		icon: null,
+		clearAt: null,
+	})
+}
+
+const save = async () => {
+	const isSuccess = await userStatusStore.saveUserStatus(userStatus.value)
+	if (isSuccess) {
+		emit('submit')
+	}
+}
+
+const revertStatus = async () => {
+	await userStatusStore.revertUserStatusFromBackup()
+	backupStatus.value = null
+	userStatus.value = { ...userStatusStore.userStatus }
+	isDirty.value = true
+}
+</script>
+
+<template>
+	<div class="user-status-form">
+		<div class="user-status-form__section">
+			<h3>{{ t('talk_desktop', 'Online status') }}</h3>
+			<UserStatusFormStatusType :status="userStatus.status" @update:status="patchStatus({ status: $event })" />
+		</div>
+
+		<div class="user-status-form__section">
+			<h3>{{ t('talk_desktop', 'Custom status') }}</h3>
+
+			<NcNoteCard v-if="backupStatus" type="info">
+				{{ t('talk_desktop', 'Your status was set automatically') }}
+			</NcNoteCard>
+
+			<UserStatusFormCustomMessage class="user-status-form__row"
+				:disabled="backupStatus"
+				:message="userStatus.message"
+				:icon="userStatus.icon"
+
+				@update:message="patchStatus({ message: $event })"
+				@update:icon="patchStatus({ icon: $event })" />
+
+			<UserStatusFormClearAt class="user-status-form__row"
+				:disabled="!statusIsUserDefined || backupStatus"
+				:clear-at="userStatus.clearAt"
+				@update:clearAt="patchStatus({ clearAt: $event })" />
+
+			<NcButton class="user-status-form__row"
+				type="secondary"
+				wide
+				:disabled="!statusIsUserDefined"
+				@click="clearUserStatusCustomMessage">
+				{{ t('talk_desktop', 'Clear custom status') }}
+			</NcButton>
```

Comment:
I think it would be better if it is placed at the bottom because it also clears predefined status.